hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleStakingInfo.java

```diff
@@ -229,4 +277,23 @@ public long[] getRewardSumHistory() {
 	public void setRewardSumHistory(final long[] rewardSumHistory) {
 		this.rewardSumHistory = rewardSumHistory;
 	}
+
+	public void clearRewardSumHistory() {
+		Arrays.fill(rewardSumHistory, 0);
+	}
+
+	public void updateRewardSumHistory(final double rewardRate, final long totalStakedRewardStart) {
+		rewardSumHistory = Arrays.copyOf(rewardSumHistory, rewardSumHistory.length);
```

Comment:
Since `copy()` is sharing the `long[] rewardSumHistory` between copies, we need to create our own before modifying it. ⚠️ The same should be done above in `clearRewardSumHistory()`.

Also, we should use the `throwIfImmutable()` check at the start of these two methods.